src/Lower.cpp

```diff
@@ -240,6 +245,7 @@ Stmt lower(const vector<Function> &outputs, const string &pipeline_name, const T
         debug(2) << "Lowering after removing varying attributes:\n" << s << "\n\n";
     }
 
+    s = simplify(s);
```

Comment:
There's another simplify call 3 lines down after remove dead allocations/trivial for loops, is this really needed?